### DISCUSSION TITLE
Get data from Wikidata

### DISCUSSION THREAD

**Discussion by ghost** [organicmaps/organicmaps#8782] [Q&A] [answered]
For #8744 I thought I'd get the data from Wikidata. Is it already possible with OM to retrieve data from Wikidata and is this a good idea for the offline side of the application?

**Comment by hb0nd** [comment, marked as answer]
[organicmaps/wikiparser](https://github.com/organicmaps/wikiparser) would need to be extended to download the Wikidata item properties, then it can be stored as usual by the generator in a `population=*` tag. this wikidata property download would be very useful for other tasks too :)

**Reply by ghost** [reply]
Thanks dude ^^.

**Reply by biodranik** [reply]
Would it be better to make a separate script that will upload missing population tags extracted from Wikidata once into OSM? Then anyone will benefit, not only OM users.

@Zverik @map-per WDYT?

**Reply by map-per** [reply]
That should be possible for CC0 licensed wikidata items. I have never done an import but the steps are explained in (1).

(1) https://wiki.openstreetmap.org/wiki/Automated_Edits_code_of_conduct

**Reply by hb0nd** [reply]
Yup, see https://wiki.openstreetmap.org/wiki/Wikidata#Importing_data_from_Wikidata_into_OSM

**Reply by Mikayex** [reply]
I'm also interested in the subject (also for France)
It seems the osm wiki recommends that the data consumers get the data from wikidata themselves: https://wiki.openstreetmap.org/w/index.php?title=Key:population#Alternatives
That way it would avoid any licensing problem with osm database.
Of course, having the data in the osm database would be better!

I don't know what is contained in Wikimedia dumps, but my idea was to build a database (SQLite?) while parsing the dumps and injecting the data found in the generation phase if no osm data is available (as a fallback).

**Comment by biodranik** [comment]
How many OpenStreetMap.org objects of type city/town/village/hamlet have Wikidata tag set but do not have population tag set?

**Reply by ghost** [reply]
For France too many for Overpass :)

![image](https://github.com/user-attachments/assets/de329880-8256-4908-b5d9-9625d781ef83)

**Reply by ghost** [reply]
But, for the opposite, I have this

![image](https://github.com/user-attachments/assets/ddd0e9e8-f90f-40b4-985c-3ea286739f72)

**Reply by map-per** [reply]
For the failed query you can try to increase the `timeout` value in line 6
